Insights/InsightsTests/InsightsTests.swift

```diff
@@ -2,35 +2,89 @@
 //  InsightsTests.swift
 //  InsightsTests
 //
-//  Created by Robert Mogos on 04/05/2018.
 //  Copyright © 2018 Algolia. All rights reserved.
 //
 
 import XCTest
 @testable import Insights
 
 class InsightsTests: XCTestCase {
-    
-    override func setUp() {
-        super.setUp()
-        // Put setup code here. This method is called before the invocation of each test method in the class.
+  
+  override func setUp() {
+    super.setUp()
+  }
+  
+  override func tearDown() {
+    super.tearDown()
+  }
+  
+  func testInitShouldFail() {
```

Comment:
I think this test doesn't really test if it fails or not:

1- If the try statement fails, it will catch and ofc the error will not be nill
2- If the try statement succeeds, it will not catch and then will go to the end of the function and the UT will succeed since we're not checking for anything, although the test should not pass in this case.

Reply:
OK